src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -132,115 +224,871 @@ static DamageTable D_8092C140 = {
+                    }
+                } else {
+                    if (this->actor.colChkInfo.damageEffect == 2) {
+                        this->effectAlpha = 4.0f;
+                        this->effectScale = 0.65f;
+                        this->drawEffect = 0;
+                    } else if (this->actor.colChkInfo.damageEffect == 4) {
+                        EnIk_HitArmor(this, globalCtx);
+                    }
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_DAMAGE);
+                    if (this->actor.colChkInfo.health != 0) {
+                        EnIk_SetupReactToAttack(this, 1);
+                    } else {
+                        EnIk_SetupCutscene(this);
+                    }
+                }
+            } else {
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_ARMOR_HIT);
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_DAMAGE);
+                if (this->actor.colChkInfo.damageEffect == 4) {
+                    EnIk_HitArmor(this, globalCtx);
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_80929BEC.s")
+void EnIk_ArmorDestroyed(EnIk* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 i;
+    EnIkStruct* ptr;
+    Vec3f effectPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_80929C80.s")
+    for (i = 0; i < ARRAY_COUNT(this->unk_550); i++) {
+        ptr = &this->unk_550[i];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_80929D04.s")
+        if (ptr->unk_24 != 0) {
+            Math_Vec3f_Sum(&ptr->unk_04, &ptr->unk_10, &ptr->unk_04);
+            ptr->unk_10.y += -1.5f;
+            if (ptr->unk_04.y < this->actor.floorHeight) {
+                ptr->unk_24 = 0;
+                ptr->unk_04.y = this->actor.floorHeight;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_80929E2C.s")
+                for (i = 0; i < 4; i++) {
+                    effectPos.x = randPlusMinusPoint5Scaled(20.0f) + ptr->unk_04.x;
+                    effectPos.y = Rand_ZeroFloat(20.0f) + ptr->unk_04.y;
+                    effectPos.z = randPlusMinusPoint5Scaled(20.0f) + ptr->unk_04.z;
+                    func_800B3030(globalCtx, &effectPos, &gZeroVec3f, &gZeroVec3f, 40, 7, 2);
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_80929E88.s")
+                SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &ptr->unk_04, 11, NA_SE_EN_EXTINCT);
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_80929F20.s")
+void EnIk_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnIk* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_80929FC8.s")
+    if (this->actionFunc != EnIk_PlayCutscene) {
+        EnIk_UpdateDamage(this, globalCtx);
+    } else {
+        this->colliderTris.base.acFlags &= ~0x80;
```

Comment:
```suggestion
        this->colliderTris.base.acFlags &= ~AC_BOUNCED;
```
Err... I think that's the right flag? not sure I've ever used it before.